builder/comp-builder.nix

```diff
@@ -143,9 +146,38 @@ stdenv.mkDerivation ({
     # The directory containing the haddock documentation.
     # `null' if no haddock documentation was built.
     haddockDir = if doHaddock' then "${docdir drv.doc}/html" else null;
-    run = runCommand (fullName + "-run") {} ''
-      ${toString component.testWrapper} ${drv}/${installedExe} | tee $out
-    '';
+
+    # This run derivation can be used to execute test, benchmark (or even
+    # exe) components.  The $out of the derivation is a file containing
+    # the resulting stdout output.
+    run = stdenv.mkDerivation ({
+      name = (fullName + "-run");
+
+      src = drv;
+
+      passthru = {
+        inherit (package) identifier;
+        config = component;
+        inherit configFiles executableToolDepends cleanSrc;
+        env = shellWrappers;
+      };
+
+      inherit (drv) meta LANG LC_ALL;
+
+      inherit doCheck doCrossCheck;
+
+      phases = ["checkPhase"];
+
+      checkPhase = ''
+        runHook preCheck
+
+        ${toString component.testWrapper} $src/${installedExe} ${lib.concatStringsSep " " component.testFlags} | tee $out
```

Comment:
Is there a good reason that `installedExe` is not `bin/${componentId.cname}`?

Reply:
I am not sure.  See
https://github.com/input-output-hk/haskell.nix/blob/6e95ba56f10634378236076575bef0a72e066a31/builder/comp-builder.nix#L120-L123

Reply:
Yeah I think it should be `bin`. Having the component name as a directory is inconvenient.